### PR TITLE
New structure for Documentation

### PR DESCRIPTION
**Pull request by agriffard**
Move Readme.md from each module to docs folder.

### CODE REVIEW

**Comment by jptissot** [comment]
Awesome !

File: mkdocs.yml [hunk 1]
```diff
@@ -29,69 +29,64 @@ markdown_extensions:
...
-        - Tenants: OrchardCore.Modules/OrchardCore.Tenants/README.md
-        - Workflows: OrchardCore.Modules/OrchardCore.Workflows/README.md
-        - DataProtection (Azure Storage): OrchardCore.Modules/OrchardCore.DataProtection.Azure/README.md
-    - Authentication Modules:
-        - Microsoft: OrchardCore.Modules/OrchardCore.Microsoft.Authentication/README.md
-        - Facebook: OrchardCore.Modules/OrchardCore.Facebook/README.md
-        - Twitter: OrchardCore.Modules/OrchardCore.Twitter/README.md
-        - GitHub: OrchardCore.Modules/OrchardCore.GitHub/README.md
-        - Google: OrchardCore.Modules/OrchardCore.Google/README.md
+        - Create a modular application: docs/guides/create-modular-application-mvc/index.md
```
Comment (sebastienros) [inline mkdocs.yml]: A reason to remove 'ing' ?

File: mkdocs.yml [hunk 2]
```diff
@@ -29,69 +29,64 @@ markdown_extensions:
...
+        - Run code on startup: docs/guides/run-code-on-startup/index.md
+        - Create a website: docs/guides/create-cms-application/index.md
+        - Add an item to the admin menu: docs/guides/add-admin-menu/index.md
+        - Install localization files: docs/guides/install-localization-files/index.md
+    - Modules:
+        - Contents: docs/modules/Contents/README.md
```
Comment (sebastienros) [inline mkdocs.yml]: Some lines are missing, any reason?
Like Title.

File: mkdocs.yml [hunk 3]
```diff
@@ -29,69 +29,64 @@ markdown_extensions:
...
-        - Facebook: OrchardCore.Modules/OrchardCore.Facebook/README.md
-        - Twitter: OrchardCore.Modules/OrchardCore.Twitter/README.md
-        - GitHub: OrchardCore.Modules/OrchardCore.GitHub/README.md
-        - Google: OrchardCore.Modules/OrchardCore.Google/README.md
+        - Create a modular application: docs/guides/create-modular-application-mvc/index.md
+        - Run code on startup: docs/guides/run-code-on-startup/index.md
+        - Create a website: docs/guides/create-cms-application/index.md
+        - Add an item to the admin menu: docs/guides/add-admin-menu/index.md
+        - Install localization files: docs/guides/install-localization-files/index.md
+    - Modules:
```
Comment (sebastienros) [inline mkdocs.yml]: Does it make sense to have "Modules" and "Core" when they are all modules? Just that some are for CMS and the others not.

Comment (jptissot) [reply mkdocs.yml]: A tag on the menu name could work here.

File: mkdocs.yml (hunk 1 above)
Comment (agriffard) [reply mkdocs.yml]: To have it on one line in the left menu on RTD

File: mkdocs.yml (hunk 2 above)
Comment (agriffard) [reply mkdocs.yml]: I am trying to make a Content Parts page that regroups all the parts.

**Comment by agriffard** [comment]
I would like to reorganize the menu items, with more main sections like these ones:

- About
  - Framework
  - Modules
  - Development
- Getting started
  - Examples
  - Faq
  - Guides
- Core
  - Admin
  - Configuration
  - Features
  - Settings
- Content management
  - Types definition
  - Parts, Fields
  - Widgets, Layers
- Theming
  - Templates
  - Liquid
  - Placement
- Search
  - Indexing
  - Lucene
  - Queries
  - SQL
- Navigation
  - Menu
  - Admin
- Assets
  - Resource
  - Media
  - Azure
- Localization
  - Translations
  - Content Localization
- Security
  - Roles, Users, Permissions
  - Open Id
  - Authentication
- Data
  - YesSql
  - GraphQL
- Tenants
- Workflows

It seems that pages are not required anymore to appear in the mkdocs.yml to be generated so we could make a shorter left menu and more level of hierarchy in the pages.

**Comment by sebastienros** [comment]
Great, I also wanted to do that, with even more details, like the paragraphs in the documents so anyone would just have to write content.

**Comment by agriffard** [comment]
New main sections as Tabs: About, Getting Started, How-to guides, Key Topics, Reference

**Comment by agriffard** [comment]
See this article to understand why we organize the documentation like this: https://www.divio.com/blog/documentation/

**Comment by sebastienros** [comment]
Ready to go ? I assume we can iterate on it later on if we need to do some changes, as you did the biggest part

**Comment by agriffard** [comment]
Ready.

**Pull request merged**